modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/portlet/action/UpdateKBArticleMVCActionCommand.java

```diff
@@ -332,19 +335,24 @@ else if (cmd.equals(Constants.UPDATE)) {
 			long[] removeFileEntryIds = ParamUtil.getLongValues(
 				actionRequest, "removeFileEntryIds");
 
-			kbArticle = _kbArticleService.updateKBArticle(
-				resourcePrimKey, title, content, description, sections,
-				sourceURL, displayDate, expirationDate, reviewDate,
-				selectedFileNames, removeFileEntryIds, serviceContext);
+			if (workflowAction == WorkflowConstants.ACTION_SAVE_DRAFT) {
+				kbArticle = _kbArticleService.updateKBArticle(
+					resourcePrimKey, title, content, description, sections,
+					sourceURL, displayDate, expirationDate, reviewDate,
+					selectedFileNames, removeFileEntryIds, serviceContext);
+			}
+			else {
+				kbArticle = _kbArticleService.updateAndUnlockKBArticle(
+					resourcePrimKey, title, content, description, sections,
+					sourceURL, displayDate, expirationDate, reviewDate,
+					selectedFileNames, removeFileEntryIds, serviceContext);
+			}
```

Comment:
Would it be simpler to do always update and just unlock when action is not ACTION_SAVE_DRAFT? This way the method `updateAndUnlockKBArticle` is not needed. I would also avoid the method name `updateAndUnlockKBArticle` because it seems that the regular `update` doesn't do the unlock (it does, not always, but it does)

```java
			kbArticle = _kbArticleService.updateKBArticle(
				resourcePrimKey, title, content, description, sections,
				sourceURL, displayDate, expirationDate, reviewDate,
				selectedFileNames, removeFileEntryIds, serviceContext);
				
			if (workflowAction != WorkflowConstants.ACTION_SAVE_DRAFT) {
				_kbArticleService.unlockKBArticle(resourcePrimKey);
			}
```

Reply:
please see https://github.com/liferay-lima/liferay-portal/pull/5080#discussion_r1462053434

Reply:
OK, if @adolfopa likes it go ahead